src/workflow_components/composite_action.py

```diff
@@ -108,6 +113,11 @@ def from_dict(obj_dict) -> "CompositeAction":
         ca = CompositeAction(name=obj_dict.get("name"), path=obj_dict["path"])
 
         ca.url = obj_dict["url"]
+
+        # Optional properties
+        if "tag" in obj_dict:
```

Comment:
Why is it a `tag` and not a `ref`? what do we expect to see here? `v1`/`main` or sha value?

Reply:
v1/main, I  have decided to save the actions by their associated tags rather than the SHA hashes they point to. Saving them in sha will require a bigger change in the way we handle data in the Redis.

Reply:
Understood.
So what is the reason we have the function to fetch ref for a tag?

Reply:
Also, we need to think whether we'll have relations issues in the graph when we are working solely with tags and not ref.
Maybe we need to write few lines of design principles for that. For how we see the solution